ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -118,16 +146,40 @@ queryEncodeNodeToClient codecConfig queryVersion blockVersion (SomeSecond query)
           throw $ QueryEncoderUnsupportedQuery (SomeSecond query) queryVersion
 
     -- From version 1 onwards, we use normal constructor tags
-    QueryVersion1 ->
+    _ ->
       case query of
         BlockQuery blockQuery ->
-            encodeListLen 2
-         <> encodeWord8 0
-         <> encodeBlockQuery blockQuery
+          requireVersion QueryVersion1 queryVersion $ mconcat
+            [ encodeListLen 2
+            , encodeWord8 0
+            , encodeBlockQuery blockQuery
+            ]
+
         GetSystemStart ->
-            encodeListLen 1
-         <> encodeWord8 1
+          requireVersion QueryVersion1 queryVersion $ mconcat
+            [ encodeListLen 1
+            , encodeWord8 1
+            ]
+
+        GetChainBlockNo ->
+          requireVersion QueryVersion2 queryVersion $ mconcat
+            [ encodeListLen 1
+            , encodeWord8 2
+            ]
+
+        GetChainPoint ->
+          requireVersion QueryVersion2 queryVersion $ mconcat
+            [ encodeListLen 1
+            , encodeWord8 3
+            ]
+
   where
+    requireVersion :: QueryVersion -> QueryVersion -> a -> a
+    requireVersion expectedVersion actualVersion a =
```

Comment:
`queryVersion` is in scope here, so you can `let actualVersion = queryVersion` instead of repeatedly passing it as a parameter.

Reply:
Fixed!